examples/history/index.js

```diff
@@ -0,0 +1,146 @@
+
+import { Value } from 'slate'
+import { Editor } from 'slate-react'
+
+import React from 'react'
+import initialValue from './value.json'
+
+/**
+ * Toolbar button component.
+ *
+ * @type {Function}
+ */
+
+const ToolbarButton = props => (
+  <span className="button" onMouseDown={props.onMouseDown}>
+    <span className="material-icons">{props.icon}</span>
+  </span>
+)
+
+/**
+ * The history stack example.
+ *
+ * @type {Component}
+ */
+
+class History extends React.Component {
+
+  /**
+   * Deserialize the initial editor value.
+   *
+   * @type {Object}
+   */
+
+  state = {
+    value: Value.fromJSON(initialValue)
+  }
+
+  /**
+   * On change.
+   *
+   * @param {Change} change
+   */
+
+  onChange = ({ value }) => {
+    this.setState({ value })
+  }
+
+  /**
+   * On delete selected text.
+   *
+   */
+
+  onClickDelete = (event) => {
+    event.preventDefault()
+    const { value } = this.state
+    const change = value.change().delete()
+    this.onChange(change)
+  }
+
+  /**
+   * On redo in history.
+   *
+   */
+
+  onClickRedo = (event) => {
+    event.preventDefault()
+    const { value } = this.state
+    const change = value.change().redo()
+    this.onChange(change)
+  }
+
+  /**
+   * On undo in history.
+   *
+   */
+
+  onClickUndo = (event) => {
+    event.preventDefault()
+    const { value } = this.state
+    const change = value.change().undo()
+    this.onChange(change)
+  }
+
+  /**
+   * Render the editor.
+   *
+   * @return {Component} component
+   */
+
+  render() {
+    return (
+      <div className="editor">
+        {this.renderToolbar()}
+        {this.renderEditor()}
+      </div>
+    )
+  }
+
+  /**
+   * Render the toolbar.
+   *
+   * @return {Element}
+   */
+
+  renderToolbar = () => {
+    const { value } = this.state
+    return (
+      <div className="menu toolbar-menu">
+        <ToolbarButton icon="delete" onMouseDown={this.onClickDelete} />
```

Comment:
I don't think we need this button because a user can just use <kbd>backspace</kbd> or <kbd>delete</kbd> to achieve the same behavior? I'd rather keep it super simple.

Reply:
Fixed!